### PR TITLE
env_process: removes message function from Exception class

### PR DESCRIPTION
**Pull request by mcasquer**
env_process: removes message function from test exception

No message function for Exception class with python3

Signed-off-by: mcasquer <mcasquer@redhat.com>
ID: 2034135

### CODE REVIEW

**Comment by mcasquer** [comment]
After applying the patch and with ksmtuned **not** installed:

 (1/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.ksm_base.disable.other_host.q35: CANCEL: Error, ksmtuned command not found. (3.19 s)
 (2/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.ksm_base.base.q35: CANCEL: Error, ksmtuned command not found. (3.21 s)
 (3/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.ksm_base.negative.q35: CANCEL: Error, ksmtuned command not found. (3.02 s)

**Comment by mcasquer** [comment]
@huangyum @PaulYuuu 

Could you pls help to rewiew the PR? Thanks!

**Review by PaulYuuu — APPROVED** [review]

**Comment by huangyum** [comment]
Suggest commit summary as: "utils_misc: cancel test case if....".

**Comment by mcasquer** [comment]
> Suggest commit summary as: "utils_misc: cancel test case if....".

Done!

File: virttest/utils_misc.py [hunk 1]
```diff
@@ -3025,7 +3025,7 @@ def get_ksmtuned_pid(self):
         try:
             utils_path.find_command("ksmtuned")
         except utils_path.CmdNotFoundError:
-            raise KSMTunedNotSupportedError
+            raise exceptions.TestCancel("Error, ksmtuned command not found.")
```
Comment (huangyum) [inline virttest/utils_misc.py:3028]: Would you pls remove 'Error' in the message? It's kind of confusing to have both 'CANCEL' and 'Error' in the result. Others look good to me.

Comment (mcasquer) [reply virttest/utils_misc.py]: Done!

**Review by huangyum — APPROVED** [review]
ACK

File: virttest/utils_misc.py [hunk 2]
```diff
@@ -3025,7 +3025,7 @@ def get_ksmtuned_pid(self):
         try:
             utils_path.find_command("ksmtuned")
         except utils_path.CmdNotFoundError:
-            raise KSMTunedNotSupportedError
```
Comment (YongxueHong) [inline virttest/utils_misc.py:3028]: I suggest that add the step to cancel it in the upper caller, like:
test_setup.py
```
        # Check if ksmtuned is running before the test
        try:
            self.ksmtuned_process = self.ksmctler.get_ksmtuned_pid()
        except utils_misc.KSMTunedNotSupportedError:
            raise exceptions.TestCancel("ksmtuned command not found.")
```
I think this API should not be related with test case level, it provides such method for caller, but the behaviours about how to handle the error should be determined by the upper caller, and we can not make sure that all the test case scenarios can cancel test due to command not found.
For the previous design(`raise KSMTunedNotSupportedError`), we can keep the method independence which doesn't coupling with other upper level.

Comment (YongxueHong) [reply virttest/utils_misc.py:3028]: Hi, @yanan-fu 
Would you like to help to review it? Thanks.

Comment (yanan-fu) [reply virttest/utils_misc.py:3028]: +1, it is test_setup's response as my understanding.

Comment (yanan-fu) [reply virttest/utils_misc.py:3028]: Although, one more high level checking is in env_process:
https://github.com/avocado-framework/avocado-vt/blob/62ae82c9848dd8a102e7b3de9b166c03268d237b/virttest/env_process.py#L989

cancel test with the test env is not suitable for our testing,  i think env_process is better than test_setup and then utils_misc.

Comment (yanan-fu) [reply virttest/utils_misc.py:3028]: @YongxueHong  What's your opinion about this ?

Comment (mcasquer) [reply virttest/utils_misc.py:3028]: @yanan-fu 

So only adding in [ksm_base.cfg](https://github.com/autotest/tp-qemu/blob/master/qemu/tests/cfg/ksm_base.cfg) the entry: `cmds_installed_host = "ksmtuned"` will be valid?

I've tested it and if ksmtuned is not installed reports the following:

 (1/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.ksm_base.disable.other_host.q35: ERROR: 'CmdNotFoundError' object has no attribute 'message' (0.20 s)

Comment (yanan-fu) [reply virttest/utils_misc.py:3028]: The reason is python3 `Exception` does not support `message` function.  This is another issue not related with this patch directly,  fix in env_process, you can try:
```
    if params.get("cmds_installed_host"):
        for cmd in params.get("cmds_installed_host").split():
            try:
                path.find_command(cmd)
            except path.CmdNotFoundError as msg:
                raise exceptions.TestSkipError(msg)
```

Comment (mcasquer) [reply virttest/utils_misc.py]: Changed! Will update test results after applying both patches

**Comment by yanan-fu** [comment]
@YongxueHong @huangyum @PaulYuuu  Could you please help re-review with the latest change ? Thanks!

**Comment by YongxueHong** [comment]
> @YongxueHong @huangyum @PaulYuuu Could you please help re-review with the latest change ? Thanks!

Hi @yanan-fu 
It looks good to me.
BTW, Would you like to explain the reason why to remove message function?
Thanks.

**Comment by yanan-fu** [comment]
> > @YongxueHong @huangyum @PaulYuuu Could you please help re-review with the latest change ? Thanks!
> 
> Hi @yanan-fu It looks good to me. BTW, Would you like to explain the reason why to remove message function? Thanks.

It is a python3 change,  `Exception` does not support message function any more. You can refer the comment above, it is cause the error that Mario commented.

**Comment by YongxueHong** [comment]
> > > @YongxueHong @huangyum @PaulYuuu Could you please help re-review with the latest change ? Thanks!
> > 
> > 
> > Hi @yanan-fu It looks good to me. BTW, Would you like to explain the reason why to remove message function? Thanks.
> 
> It is a python3 change, `Exception` does not support message function any more. You can refer the comment above, it is cause the error that Mario commented.

Hi @yanan-fu 
Sorry, this question is not for you, I mean @mcasquer 
Hi @mcasquer 
Would you like to explain the reason why to remove message function in your commit message?
Thanks.

**Comment by huangyum** [comment]
> > > @YongxueHong @huangyum @PaulYuuu Could you please help re-review with the latest change ? Thanks!
> > 
> > 
> > Hi @yanan-fu It looks good to me. BTW, Would you like to explain the reason why to remove message function? Thanks.
> 
> It is a python3 change, `Exception` does not support message function any more. You can refer the comment above, it is cause the error that Mario commented.

Does it work with python2 ?

**Comment by yanan-fu** [comment]
> Does this change work with python2 ?

@huangyum As far as i know, it still can work with python.  And, avocado framework does not suport python2  officially any more,  we can ignore the support for python2.

**Comment by mcasquer** [comment]
> Would you like to explain the reason why to remove message function in your commit message?

@YongxueHong 

Sorry, I misunderstood the message :grimacing:
Commit message updated!

**Comment by mcasquer** [comment]
After applying https://github.com/autotest/tp-qemu/pull/3247 patch and the one from this PR, with ksmtuned **not** installed:

**ksm_base**
`(1/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.x86_64.io-github-autotest-qemu.ksm_base.disable.other_host.q35: SKIP: Command 'ksmtuned' could not be found in any of the PATH dirs: ['/bin', '/root/.local/bin', '/root/bin', '/sbin', '/usr/bin', '/usr/libexec', '/usr/local/bin', '/usr/local/sbin', '/usr/sbin']`

`(2/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.x86_64.io-github-autotest-qemu.ksm_base.base.q35: SKIP: Command 'ksmtuned' could not be found in any of the PATH dirs: ['/bin', '/root/.local/bin', '/root/bin', '/sbin', '/usr/bin', '/usr/libexec', '/usr/local/bin', '/usr/local/sbin', '/usr/sbin']`

`(3/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.x86_64.io-github-autotest-qemu.ksm_base.negative.q35: SKIP: Command 'ksmtuned' could not be found in any of the PATH dirs: ['/bin', '/root/.local/bin', '/root/bin', '/sbin', '/usr/bin', '/usr/libexec', '/usr/local/bin', '/usr/local/sbin', '/usr/sbin']`
 
 If installed ksm_base pass as expected:
 
`(1/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.x86_64.io-github-autotest-qemu.ksm_base.disable.other_host.q35: PASS (358.06 s)`

`(2/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.x86_64.io-github-autotest-qemu.ksm_base.base.q35: PASS (341.82 s)`

`(3/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.x86_64.io-github-autotest-qemu.ksm_base.negative.q35: PASS (344.22 s)`

**Comment by mcasquer** [comment]
@YongxueHong 

Could you please give ACK and workflow approval? Thanks!

**Review by YongxueHong — APPROVED** [review]

**Review by yanan-fu — APPROVED** [review]
Acked-by: Yanan Fu <yfu@redhat.com>

**Comment by YongxueHong** [comment]
Hi @huangyum @PaulYuuu 
Do you have comments for the latest update?
Thanks.

**Review by huangyum — APPROVED** [review]
ACK

**Review by PaulYuuu — APPROVED** [review]

**Comment by YongxueHong** [comment]
Thanks all, let's merge it.

**Pull request merged**